### PR TITLE
kvm/: various fixes, preparing for ELF and 64-bit support

### PR DESCRIPTION
**Pull request by rminnich**
This is the first step in adding ELF file and 64-bit support.

Add document comments to kvm.go as suggested by several linters

add ioctl support helpers to generate ioctl commands. This reduces the number of magic constants

Add support for KVM_TRANSLATE. This is incredibly helpful when debugging VM startup, to find mistakes in page tables.

Add an ExitType and a stringer for it.

The stringer was generated by the Go stringer, which still has lots of bugs around use of the errors package, necessitating a workaround in how it was generated. Hence the comment in kvm.go about it.

Use the stringer in machine/machine.go so users can see an error string, not a machine number.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

**Comment by rminnich** [comment]
Note that with a second PR after this one, to machine/, I am able to boot ELF kernels, 64- and 32-bit, as well as bzimage.

**Comment by rminnich** [comment]
This passes coverall and all tests until just now, travis is having "problems" today it seems

**Comment by rminnich** [comment]
dammit. These build failures are some sort of travis problem. Maybe it's time to try somewhere else.

**Review by bobuhiro11 — COMMENTED** [review]
Thanks for the improvement. It looks like Travis CI is failing in unrelated areas such as package installation. I will rerun CI after some time.

File: kvm/exittype_string.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	_ = x[EXITS390SIEIC-13]
+	_ = x[EXITS390RESET-14]
+	_ = x[EXITDCR-15]
+	_ = x[EXITNMI-16]
+	_ = x[EXITINTERNALERROR-17]
+	_ = x[EXITIOIN-0]
+	_ = x[EXITIOOUT-1]
+}
+
+// This was generated by the stringer program as a const. golangci-lint hated the long line it generated.
```
[the remark this answers is not in the document]
Comment (rminnich) [reply kvm/exittype_string.go:31]: thanks. I hope we can get this in, I'd like to get the 64-bit support in once this goes in :-)

**Review by bobuhiro11 — APPROVED** [review]

**Comment by bobuhiro11** [comment]
Looks good to me. Thanks.

**Pull request merged**